tests/components/yessssms/test_notify.py

```diff
@@ -151,210 +149,219 @@ async def test_connection_error_on_init(hass, caplog, valid_settings, connection
+
+    assert requests_mock.called is True
+    assert requests_mock.call_count == 4
+    assert (
+        requests_mock.last_request.scheme
+        + "://"
+        + requests_mock.last_request.hostname
+        + requests_mock.last_request.path
+        + "?"
+        + requests_mock.last_request.query
+    ) in yessssms.yesss._logout_url  # pylint: disable=protected-access
+
+
+async def test_no_recipient_error(yessssms, caplog):
+    """Test for missing/empty recipient."""
+    message = "Testing YesssSMS platform :)"
+    # pylint: disable=protected-access
+    yessssms._recipient = ""
+
+    with caplog.at_level(logging.ERROR):
+        yessssms.send_message(message)
+    for record in caplog.records:
+        if (
+            record.levelname == "ERROR"
+            and record.name == "homeassistant.components.yessssms.notify"
+        ):
+            assert (
+                "You need to provide a recipient for SMS notification" in record.message
+            )
+
+
+async def test_sms_sending_error(yessssms, requests_mock, caplog):
+    """Test sms sending error."""
+    requests_mock.post(
+        # pylint: disable=protected-access
+        yessssms.yesss._login_url,
+        status_code=302,
+        # pylint: disable=protected-access
+        headers={"location": yessssms.yesss._kontomanager},
+    )
+    # pylint: disable=protected-access
+    login = yessssms.yesss._logindata["login_rufnummer"]
+    requests_mock.get(
+        # pylint: disable=protected-access
+        yessssms.yesss._kontomanager,
+        status_code=200,
+        text=f"test...{login}</a>",
+    )
+    requests_mock.post(
+        # pylint: disable=protected-access
+        yessssms.yesss._websms_url,
+        status_code=HTTP_INTERNAL_SERVER_ERROR,
+    )
+
+    message = "Testing YesssSMS platform :)"
+
+    with caplog.at_level(logging.ERROR):
+        yessssms.send_message(message)
+
+    assert requests_mock.called is True
+    assert requests_mock.call_count == 3
+    for record in caplog.records:
+        if (
+            record.levelname == "ERROR"
+            and record.name == "homeassistant.components.yessssms.notify"
+        ):
+            assert "YesssSMS: error sending SMS" in record.message
+
+
+async def test_connection_error(yessssms, requests_mock, caplog):
+    """Test connection error."""
+    requests_mock.post(
+        # pylint: disable=protected-access
+        yessssms.yesss._login_url,
+        exc=yessssms.yesss.ConnectionError,
+    )
+
+    message = "Testing YesssSMS platform :)"
+
+    with caplog.at_level(logging.ERROR):
+        yessssms.send_message(message)
+
+    assert requests_mock.called is True
+    assert requests_mock.call_count == 1
+    for record in caplog.records:
+        if (
+            record.levelname == "ERROR"
+            and record.name == "homeassistant.components.yessssms.notify"
+        ):
+            assert "cannot connect to provider" in record.message
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6285789)